### ISSUE TITLE
Geometric.DisplayPointsFromArray

### ISSUE THREAD

**Issue by DTemeroso** [ichim/LeafletForBlazor-NuGet#20]
When you're using Geometric.DisplayPointsFromArray, can't i use another type of point instead of the PointSymbol?

Because when I use PointIcon I get an error message.

![image](https://github.com/ichim/LeafletForBlazor-NuGet/assets/160488451/3ea30ab5-b01a-4d59-b102-5b5ff7f719f4)

**Comment by ichim** [comment]
Hello sir,

The package is under development.
At this moment, only PointSymbol symbolization is implemented. In future versions, addPoint() will also have icon symbolization available...

**Comment by DTemeroso** [comment]
ok,
What would be the best method to insert multiple markers on the map?

**Comment by DTemeroso** [comment]
when we're using the Geometric.Points.upload, add ,appearance methods, what's the order they should be put in, if you're not using buttons?

**Comment by ichim** [comment]
Appearance() and Appearance Type() have static behavior, they store the symbols in a current view.  The order of calling add() and appearance should not matter.

**Comment by ichim** [comment]
Appearance() and Appearance Type() have static behavior, they store the symbols in a current view.  The order of calling add() and appearance should not matter.
...soon I will update to version 2.0.2.6

**Comment by DTemeroso** [comment]
thank , I'll be waiting for it

**Comment by ichim** [comment]
Thank you too

**Comment by DTemeroso** [comment]
where "RealTimeMap.VisibilityZoomLevel" can be placed in the code and how it should be used

**Comment by ichim** [comment]
No sir, is a class. As a general rule, properties that can receive values have names starting with a lowercase letter (camelcase).

You can set the visibility of a type (Stream Point) using the Appearance Type() method with the pattern property

**Comment by DTemeroso** [comment]
Ok, thanks for explain

**Comment by DTemeroso** [comment]
the Esri plugins just works on the Map control?

**Comment by ichim** [comment]
Yes sir, for this moment.
Let me explain!  I made a big mistake in the design of the Map control that cannot be corrected.  That's why we started the development of the RealTimMap control.  RealTimeMap will have Map functionality.

RealTimeMap will be dynamic, capable of working with moving points...

**Comment by ichim** [comment]
... features will be introduced gradually.  The way the new version is developed is slightly more complicated than the old version. Sorry

**Comment by ichim** [comment]
Hello sir,
I uploaded a new version of the package (2.0.2.6).  It will be available in a few minutes after indexing.  In this package you will find the OnAfterMapLoaded event method.This method fires after the map is fully loaded into the page.

**Comment by DTemeroso** [comment]
Hi, 
Thanks for notifying the release of the new version

**Comment by DTemeroso** [comment]
An error is appearing for MapEventArgs
```
public async Task onLoadControl(RealTimeMap.MapEventArgs value)
{
	if (realTimeMap != null)
	{
		await realTimeMap.Geometric.Points.upload(pointsStart, true);
	}
}
``` 
![image](https://github.com/ichim/LeafletForBlazor-NuGet/assets/160488451/426f3388-9dca-4286-a907-76c90f8d5065)

**Comment by ichim** [comment]
... check if you have updated to the new version 2.0.2.6

**Comment by DTemeroso** [comment]
I updated, the problem is when I rebuild the project and run it, it returns to the previous version

**Comment by DTemeroso** [comment]
i have to restart the vsual studio

**Comment by DTemeroso** [comment]
Sorry for disturbing you

**Comment by ichim** [comment]
it's working?

**Comment by DTemeroso** [comment]
Yes
the problem was whit the visual studio

**Comment by DTemeroso** [comment]
the Apperance method can use all 3 types of point?

**Comment by ichim** [comment]
I did not understand the question. You have 3 types of StreamPoint and you want to symbolize them all the same?

**Comment by DTemeroso** [comment]
Ok , thanks

**Comment by DTemeroso** [comment]
Ichim, you work for Esri, is it possible to explain in a summarized form, the TOU to use the api, if you can, legally

**Comment by DTemeroso** [comment]
if not it's ok

**Comment by ichim** [comment]
Are you referring to the Esri API?

**Comment by DTemeroso** [comment]
yes

**Comment by ichim** [comment]
Well, how do I start? I will tell you in broad terms. Esri is a world leader in the development of GIS software solutions. It offers a complete set of tools for the analysis and development of very high performance software applications. In addition, it supports Open Source solutions. One of these solutions is Leaflet developed by Volodymyr Agafonkin. In this regard, Esri has developed the "open source Leaflet plug-in": 
https://developers.arcgis.com/esri-leaflet/
It's just that to use this API (Leaflet plug-in) you need to generate a API key code:
https://developers.arcgis.com/esri-leaflet/get-started/
Generating this key code is free!!
The use can be made for educational, demo and commercial purposes and is subject to the terms of the Apache License 2.0
ArcGIS API performance is clearly superior to Leaflet API, Bing Map or other solutions!

**Comment by DTemeroso** [comment]
Currently, I am developing an application for a company, the application will be used within the company itself, it is considered commercial?

**Comment by ichim** [comment]
No sir!
Commercial means that the application is used by third parties under an agreement.

**Comment by DTemeroso** [comment]
It is possible to turn a point into a "button" by clicking on the point to make appear a sidebar with custom information about that point representes.

**Comment by ichim** [comment]
To understand that you want to click on the map and get a StreamPoint?

**Comment by DTemeroso** [comment]
No. Stream.Point is already on the map and I want to click on it to show the information about that point in a sidebar

**Comment by ichim** [comment]
...a kind of ToolTips?

**Comment by DTemeroso** [comment]
Yes. Sorry for the late response

**Comment by DTemeroso** [comment]
With the new method, it is possible to do what I had asked before about, by "clicking on an existing point", make it appear in a sidebar with the information of that point.

**Comment by ichim** [comment]
If you refer to the new event methods in the release 2.0.2.8, the answare is no. Sorry

**Comment by DTemeroso** [comment]
ok,thaks

**Comment by DTemeroso** [comment]
The "useMapBounds" parameter in the Esri plugin "EsriGeocodingSearchParameters", what does it do?

**Comment by ichim** [comment]
When useMapBounds is true, the search is done only into displayed area (map). When is false, the search is done everyone.

**Comment by DTemeroso** [comment]
When using "EsriGeocodingSearchParameters" is possivel to iterate a list of address?

**Comment by ichim** [comment]
It's an interactive search.  Only the search list displays a list of addresses that match the criteria...

**Comment by DTemeroso** [comment]
So, if I use a list with addresses that I chose, I won't be able to geocode each of the addresses.Right

**Comment by ichim** [comment]
Hello sir,

Version 2.0.4.2 of the package allows configuring Tooltips for the StreamPoint collection. Configuration is done through the pattern property of the Appearance() or AppearanceOnType() method:

```
realTimeMap.Geometric.Points.AppearanceOnType(item => item.type == "intervention vehicle").pattern = new RealTimeMap.PointTooltip()
    {
        content = "<h6>${value}</h6>${type}</br>${guid}",
        opacity = 0.8,
        permanent = true
    };
```
